source/PlanetLabel.cpp

```diff
@@ -39,6 +40,41 @@ namespace {
 	const double LINE_GAP = 1.7;
 	const double GAP = 6.;
 	const double MIN_DISTANCE = 30.;
+
+	// Check if the given label for the given stellar object and direction overlaps with any other stellar object in the system.
+	bool Overlaps(const System &system, const StellarObject &object, double zoom, double width, int direction)
+	{
+		bool overlaps = false;
+
+		Point start = zoom * (object.Position() +
+			(object.Radius() + INNER_SPACE + LINE_GAP + LINE_LENGTH) * Angle(LINE_ANGLE[direction]).Unit());
+		Point unit(LINE_ANGLE[direction] > 180. ? -1. : 1., 0.);
```

Comment:
This converts from "all possible directions" to "only leftward and rightward facing vectors". For the labels that is currently fine, but maybe add a comment here to describe that this conversion is fine for this type of label.